src/protocol/attribution/aggregate_credit.rs

```diff
@@ -100,33 +261,35 @@ async fn bit_decompose_breakdown_key<F: Field>(
 }
 
 /// Sort the input by `aggregation_bit` first, then by `breakdown_key`
-#[allow(dead_code)]
 async fn sort_by_aggregation_bit_and_breakdown_key<F: Field>(
     ctx: SemiHonestContext<'_, F>,
     input: &[CappedCreditsWithAggregationBit<F>],
 ) -> Result<Vec<CappedCreditsWithAggregationBit<F>>, Error> {
-    // Sort by aggregation_bit
-    let sorted_by_aggregation_bit = sort_by_aggregation_bit(ctx.clone(), input).await?;
+    // Prepend the aggregation_bit bits with breakdown_key bits. This results
+    // in sorting the sidecar by aggregation_bit first, and then by breakdown_key.
+
+    let mut permutation_keys = vec![input
+        .iter()
+        .map(|x| x.aggregation_bit.clone())
+        .collect::<Vec<_>>()];
 
-    // Next, sort by breakdown_key
     // TODO: Change breakdown_keys to use XorReplicated to avoid bit-decomposition calls
-    let breakdown_keys = transpose(
-        &bit_decompose_breakdown_key(
-            ctx.narrow(&Step::BitDecomposeBreakdownKey),
-            &sorted_by_aggregation_bit,
-        )
-        .await?,
+    let mut breakdown_keys = transpose(
+        &bit_decompose_breakdown_key(ctx.narrow(&Step::BitDecomposeBreakdownKey), input).await?,
     );
 
+    permutation_keys.append(&mut breakdown_keys);
+
     let sort_permutation = generate_permutation_and_reveal_shuffled(
         ctx.narrow(&Step::GeneratePermutationByBreakdownKey),
-        &breakdown_keys,
-        F::Integer::BITS,
+        &permutation_keys,
+        F::Integer::BITS + 1, // breakdown_keys bits + aggregation_bit bit
```

Comment:
This is wasted work.

Assuming my maximum breakdown key is 8, then you only need to sort by the least significant 3 bits of the bit-decomposed breakdown keys. Assuming you're using Fp32BitPrime, then F::Integer::BITS is 32... so you'll be doing a radix sort on 32 bits... even though we know in advance that the list is sorted after just 3 bits and everything else is a zero.